gms/gossiper.hh

```diff
@@ -228,6 +228,7 @@ private:
 
     /* unreachable member set */
     std::unordered_map<inet_address, clk::time_point> _unreachable_endpoints;
+    uint64_t _unreachable_endpoints_version = 0;
```

Comment:
I think we do not really need to introduce _unreachable_endpoints_version. In almost every cases, we always update _live_endpoints_version and the new  _unreachable_endpoints_version together. 

Node being live and unreachable is a rare operation. It is better to make the logic simple than optimize over calling a few unreachable nodes.

Reply:
The simplest way would be to always replicate both in today's `replicate_live_endpoints_on_change` (that could be renamed to `replicate_live_and_unreachable_endpoints` and bump `_live_endpoints_version` unconditionally there.

Reply:
Yes, bumping the version but _live or _unreachable endpoints does not change is not a big deal.